src/core/ref.jl

```diff
@@ -0,0 +1,13 @@
+function ref_add_function!(ref_function::Function, it::Symbol, ref::Dict{Symbol, <:Any}, data::Dict{String, <:Any})
```

Comment:
As per previous discussion, I would consider calling this something like `apply_to_all!` or `ref_apply_to_all!`.  Also I would make the argument order here consistent with similar functions, which i think would be `ref, data, func, it`.

If you prefer this ordering convention we can consider that as well (it looks more like a function call, which is nice) but if so, all similar functions should be consistent.

Reply:
Agree on the function name change. I do prefer the current argument ordering convention, but I will make all similar functions consistent.